File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string res = "";
        int n = s.size();
        
        for(int i=0 ; i<n ; i++){
            int m = res.size();
            if(m<2) {res+=s[i]; continue;}
            if((res[m-1] != res[m-2]) || (res[m-1] != s[i])) res += s[i];
        }
        
        return res;
    }
};

File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.rs
impl Solution {
    pub fn make_fancy_string(s: String) -> String {
        let mut res = String::new();
        let mut prev_char = '\0';
        let mut count = 0;

        for c in s.chars() {
            if c == prev_char {
                count += 1;
            } else {
                count = 1;
            }

            if count < 3 {
                res.push(c);
            }
            prev_char = c;
        }

        res
    }
}
